### PR TITLE
[autoscaler] Update Autoscaler SDK

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
* Exposes Configuration options to the SDK
* Splits `ray_constants` into `ray_autoscaler_constants` for _autoscaler only_ constants.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/autoscaler/_private/autoscaler.py [hunk 1]
```diff
@@ -26,9 +26,10 @@
 from ray.autoscaler._private.util import ConcurrentCounter, validate_config, \
     with_head_node_ip, hash_launch_conf, hash_runtime_conf, \
     DEBUG_AUTOSCALING_STATUS, DEBUG_AUTOSCALING_ERROR
-from ray.ray_constants import AUTOSCALER_MAX_NUM_FAILURES, \
-    AUTOSCALER_MAX_LAUNCH_BATCH, AUTOSCALER_MAX_CONCURRENT_LAUNCHES, \
-    AUTOSCALER_UPDATE_INTERVAL_S, AUTOSCALER_HEARTBEAT_TIMEOUT_S
+from ray.autoscaler.ray_autoscaler_constants import \
```
Comment (richardliaw) [inline python/ray/autoscaler/_private/autoscaler.py:29]: `ray.autoscaler.constants`? or else it's a bit redundant?

**Review by ericl — CHANGES_REQUESTED** [review]
@ijrsvt why do we need to expose the constants? You can just copy paste them if you need the values for some reason.

File: python/ray/autoscaler/_private/autoscaler.py (hunk 1 above)
Comment (ericl) [reply python/ray/autoscaler/_private/autoscaler.py:29]: `ray.autoscaler._private.constants`

File: python/ray/autoscaler/sdk.py [hunk 2]
```diff
@@ -186,3 +186,15 @@ def _as_config_file(cluster_config: Union[dict, str]):
     if not os.path.exists(cluster_config):
         raise ValueError("Cluster config not found {}".format(cluster_config))
     return cluster_config
+
+
+def bootstrap_config(cluster_config: Dict[str, any],
+                     no_config_cache: bool = False) -> bool:
+    """Bootstrap the config file based on the NodeProvider."""
```
Comment (ericl) [inline python/ray/autoscaler/sdk.py:193]: Can you expand on these docs a bit?

File: python/ray/autoscaler/_private/autoscaler.py (hunk 1 above)
Comment (ijrsvt) [reply python/ray/autoscaler/_private/autoscaler.py:29]: SGTM, that is far better.

File: python/ray/autoscaler/sdk.py (hunk 2 above)
Comment (ijrsvt) [reply python/ray/autoscaler/sdk.py]: of course!

**Review by ericl — APPROVED** [review]

**Pull request merged**